debezium-embedded/src/test/java/io/debezium/embedded/AbstractConnectorTest.java

```diff
@@ -500,6 +503,144 @@ protected SourceRecords consumeRecordsByTopic(int numRecords, boolean assertReco
+     * @return the actual number of Dml records that were consumed
+     * @throws InterruptedException if the thread was interrupted while waiting for a record to be returned
+     */
+    protected int consumeDmlRecordsByTopic(int numberOfRecords, int breakAfterNulls, Consumer<SourceRecord> recordConsumer, boolean assertRecords)
+            throws InterruptedException {
+        int recordsConsumed = 0;
+        int nullReturn = 0;
+        Set<String> endTransactions = new LinkedHashSet<>();
+        while (recordsConsumed < numberOfRecords) {
+            SourceRecord record = consumedLines.poll(pollTimeoutInMs, TimeUnit.MILLISECONDS);
+            if (record != null) {
+                nullReturn = 0;
+                if (isTransactionRecord(record)) {
+                    final Struct value = (Struct) record.value();
+                    final String status = value.getString(TransactionMonitor.DEBEZIUM_TRANSACTION_STATUS_KEY);
+                    if (status.equals(TransactionStatus.BEGIN.name())) {
+                        endTransactions.add(value.getString(TransactionMonitor.DEBEZIUM_TRANSACTION_ID_KEY));
+                    }
+                    else {
+                        endTransactions.remove(value.getString(TransactionMonitor.DEBEZIUM_TRANSACTION_ID_KEY));
+                    }
+                }
+                else {
+                    ++recordsConsumed;
+                }
+                if (recordConsumer != null) {
+                    recordConsumer.accept(record);
+                }
+                if (Testing.Debug.isEnabled()) {
+                    Testing.debug("Consumed record " + recordsConsumed + " / " + numberOfRecords + " ("
+                            + (numberOfRecords - recordsConsumed) + " more)");
+                    debug(record);
+                }
+                else if (Testing.Print.isEnabled()) {
+                    Testing.print("Consumed record " + recordsConsumed + " / " + numberOfRecords + " ("
+                            + (numberOfRecords - recordsConsumed) + " more)");
+                    print(record);
+                }
+                if (assertRecords) {
+                    VerifyRecord.isValid(record);
+                }
+            }
+            else {
+                if (++nullReturn >= breakAfterNulls) {
+                    return recordsConsumed;
+                }
+            }
+        }
+
+        while (!endTransactions.isEmpty()) {
+            SourceRecord record = consumedLines.poll(pollTimeoutInMs, TimeUnit.MILLISECONDS);
+            if (record != null) {
+                nullReturn = 0;
+                if (isTransactionRecord(record)) {
+                    final Struct value = (Struct) record.value();
+                    final String status = value.getString(TransactionMonitor.DEBEZIUM_TRANSACTION_STATUS_KEY);
+                    if (status.equals(TransactionStatus.END.name())) {
```

Comment:
Even if it is not strictly necessary I'd prefer to process even the incoming tx `BEGIN` to ensure symmetry between BEGIN/END messages.